crates/iota-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -3482,34 +3581,62 @@ impl AuthorityPerEpochStore {
+                                    .protocol_config
+                                    .congested_objects_gas_price_feedback_mechanism()
+                                {
+                                    let current_commit_suggested_gas_price =
+                                        self.reference_gas_price();
+                                    let suggested_gas_price = previously_deferred_tx_digests
+                                        .get(certificate.digest())
+                                        .map_or_else(
+                                            || current_commit_suggested_gas_price,
+                                            |deferral_key_suggested_gas_price_pair| {
+                                                deferral_key_suggested_gas_price_pair
+                                                    .1
+                                                    .expect(
+                                                        "Suggested gas price for transactions \
+                                                        previously deferred due to congestion must \
+                                                        not be None if the gas price feedback is \
+                                                        enabled.",
+                                                    )
+                                                    .min(current_commit_suggested_gas_price)
+                                            },
+                                        );
+
+                                    Some(suggested_gas_price)
+                                } else {
+                                    None
+                                };
+
                                 if transaction_deferral_within_limit(
                                     &deferral_key,
                                     self.protocol_config()
                                         .max_deferral_rounds_for_congestion_control(),
                                 ) {
-                                    ConsensusCertificateResult::Deferred(deferral_key)
+                                    ConsensusCertificateResult::Deferred {
+                                        deferral_key,
+                                        suggested_gas_price,
+                                    }
                                 } else {
                                     // Cancel the transaction that has been deferred for too long.
 
-                                    let suggested_gas_price = shared_object_congestion_tracker
-                                        .compute_suggested_gas_price(&certificate)
-                                        .expect(
-                                            "cancelled transaction must have at least one shared \
-                                            object and calculated suggested gas price",
-                                        );
```

Comment:
Wouldn't that change the behaviour, or at least remove that check, as long as `congested_objects_gas_price_feedback_mechanism` isn't enabled?